### PR TITLE
Walkingastro

### PR DESCRIPTION
**Pull request by BastionWT**
;walkingastro.lic is designed to observe either the Sun (day) or the Heart constellation (night) on a 215 second timer. It will continue to do so until the Survival prediction pool is full (observation messaging), at which point it will check the Astrology skill's exp. If under 25, it will align to the Survival skill designated in the user's setup.yaml (astrology_prediction_skills: -> Survival), or default to "survival" as the base.yaml setting and predict the future.

It is designed for out of combat use, similar to the newly included ;play script. It is meant to be run as a background process, and will time out intentionally when certain scripts are in-process.

This script assume the usage of a telescope and that PG is up.

### CODE REVIEW

**Comment by BastionWT** [comment]
Any action on this one? Aruthas, Msaia, and I have been running it without issues for the past ~4 days since the bput changes went in.

File: walkingastro.lic [hunk 1]
```diff
@@ -0,0 +1,210 @@
+custom_require.call %w[events]
```
Comment (rpherbig) [inline walkingastro.lic]: 💡 Needs a header that links to elanthipedia

File: walkingastro.lic [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        day_center
+      end
+      observe
+      return
+    end
+  end
+
+  def night_center
+    echo "Night center"
+    fput("center telescope on heart")
```
Comment (rpherbig) [inline walkingastro.lic]: ❔ Why 'heart' in specific?

File: walkingastro.lic [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+      waitrt?
+      pause 1
+      fput("predict future")
+      waitrt?
+      pause 1
+    end
+  end
+
+  def can_observe?
+    return !['buff', 'steal', 'combat-trainer', 'pick', 'craft', 'shape', 'sew','bescort', 'remedy', 'forge', 'carve', 'performance', 'theurgy', 'hlctheurgy', 'astrology', 'astrology2', 'study-art', 'mech-lore', 'first-aid'].any? { |script| Script.running?(script) }
```
Comment (rpherbig) [inline walkingastro.lic]: I'm not a fan of this approach... this list doesn't seem exhaustive nor easy to maintain

**Comment by rpherbig** [comment]
Looks ok to me. @jandersson can you glance it over?

**Comment by rpherbig** [comment]
> Any action on this one? Aruthas, Msaia, and I have been running it without issues for the past ~4 days since the bput changes went in.

Sorry, the holiday weekend kept me busy.

File: walkingastro.lic (hunk 1 above)
Comment (BastionWT) [reply walkingastro.lic]: Righto, will do.

File: walkingastro.lic (hunk 2 above)
Comment (BastionWT) [reply walkingastro.lic]: Heart is the only Survival night time constellation available all year, with no rising/setting during the night at unspecified times. It, like the sun during the day, is always available at night. As an added bonus, it fills the same pool that the sun does, which allows for the pool to fill in a timely manner for predictions.

File: walkingastro.lic (hunk 3 above)
Comment (BastionWT) [reply walkingastro.lic]: Can you recommend an alternative? I literally harvested from the ;play script that is being used. Basically, I want walkingastro to not run during any script that can fill your hands with 1 or 2 objects, as a telescope needs both hands to center and peer.

File: walkingastro.lic (hunk 2 above)
Comment (rpherbig) [reply walkingastro.lic]: 👍 thank you for the explanation

File: walkingastro.lic (hunk 3 above)
Comment (rpherbig) [reply walkingastro.lic]: I'll have to ponder it. Since this is a new script, people are effectively opting-in to its approach, so this is fine for now.

**Pull request merged**